### PR TITLE
olm: update to 3.0.0

### PR DESCRIPTION
**Pull request by ndowens**

### CODE REVIEW

**Comment by maxice8** [comment]
Please test the python bindings, the removed patch is meant to fix the python bindings.

**Comment by ndowens** [comment]
> Please test the python bindings, the removed patch is meant to fix the python bindings.

make test : 
https://termbin.com/875h

**Comment by maxice8** [comment]
On Mon, Jan 14, 2019 at 08:41:42PM -0800, Nathan wrote:
> > Please test the python bindings, the removed patch is meant to fix the python bindings.
> 
> make test : 
> https://termbin.com/875h
> 
> 
> -- 
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/void-linux/void-packages/pull/7151#issuecomment-454265295

Sounds fine-ish, please check if you can import the olm bindings

**Comment by ndowens** [comment]
Probably someone will have to check that part, I don't know much about programming if any really :\

**Comment by maxice8** [comment]
install it, start python and try to import the python module

**Comment by maxice8** [comment]
image

**Comment by ndowens** [comment]
It doesn’t have that file to patch anymore but I MIGHT know how to fix it

**Comment by ndowens** [comment]
Going to contact upstream and see what can be done

**Comment by ndowens** [comment]
@maxice8 This worked for me, no import error(tested python3)

File: srcpkgs/olm/template [hunk 1]
```diff
@@ -1,17 +1,19 @@
...
+makedepends="python-devel python3-devel libffi-devel"
+depends="python-cffi python3-cffi python-future python3-future"
+short_desc="Implementation of the Double Ratchet cryptographic ratchet"
 maintainer="Adam Beckmeyer <adam_gpg@thebeckmeyers.xyz>"
 license="Apache-2.0"
 homepage="https://git.matrix.org/git/olm/about/"
 distfiles="https://git.matrix.org/git/olm/snapshot/${pkgname}-${version}.tar.gz"
-checksum=533714fb84860e04c185790d16ef9085f15e902c2105db941d5c7e92b0565ef8
+checksum=4c54f93e4f426a9d5c8e66e63113d42fd2afca240cdcff30a469a342a6ec3bc6
+nocross=yes
```
Comment (Hoshpak) [inline srcpkgs/olm/template]: why mark it nocross? It was possible to cross-compile before, the package is in the aarch64 repo.

Comment (ndowens) [reply srcpkgs/olm/template]: During the install stage it uses the host gcc instead of target’s

Comment (Hoshpak) [reply srcpkgs/olm/template]: An it isn't possible to find a way around that or patch it to do the right thing? Adding nocross to new packages that simply cannot be crosscompiled is one thing but setting existing packages nocross means the repositories have to be cleaned up manually and people already using a package on cross-compiled architectures will loose functionality.

Comment (ndowens) [reply srcpkgs/olm/template]: I will try more, as I was trying to avoid setting nocross

**Comment by ndowens** [comment]
I can not fix cross-compile, no matter if I try to override LD, something with cffi uses host's ld

**Comment by ndowens** [comment]
https://github.com/matrix-org/olm/issues/16 This is the bug report, so hopefully it will be fixed in the future, so this package can be updated to be cross-friendly

**Pull request closed without merging**